clang/test/SemaSYCL/Inputs/sycl.hpp

```diff
@@ -329,6 +330,33 @@ class __attribute__((sycl_special_class)) stream {
   int FlushBufferSize;
 };
 
+template <typename ElementType, access::address_space addressSpace>
+struct DecoratedType;
+
+template <typename ElementType>
+struct DecoratedType<ElementType, access::address_space::private_space> {
+  using type = __attribute__((opencl_private)) ElementType;
+};
+
+template <typename ElementType>
+struct DecoratedType<ElementType, access::address_space::generic_space> {
+  using type = ElementType;
+};
+
+template <typename ElementType>
+struct DecoratedType<ElementType, access::address_space::global_space> {
+  using type = __attribute__((opencl_global)) ElementType;
+};
+
+template <typename T, access::address_space AS> class multi_ptr {
+  using pointer_t = typename DecoratedType<T, AS>::type *;
+  pointer_t m_Pointer;
+
+public:
+  multi_ptr(T *Ptr) : m_Pointer((pointer_t)(Ptr)) {}
+  pointer_t get() { return m_Pointer; }
+};
+
 namespace ext {
```

Comment:
Could you please explain what the changes are here for my own learning? Thank you

Reply:
It is mock declaration for SYCL `multi_ptr` class that can be used for front-end testing, we have them for other SYCL classes as well. I wanted to make sure that the warning will be emitted for `multi_ptr` class defined inside system headers, since https://github.com/intel/llvm/issues/3172 reports the problem with specializations of `multi_ptr` class, so I've added it.

Reply:
Thanks for the explanation!